### DISCUSSION TITLE
Playing online audio on speakers

### DISCUSSION THREAD

**Discussion by hay-dude007** [cc-tweaked/CC-Tweaked#1207] [Q&A] [answered]
Can someone tell me how to put online audio into speakers? It says
Usage:
speaker play <file or **URL**> [speaker]

How do I play online audio immediately from the "start screen" where you type the programs.

Also i have already tried that but it just gave me static

**Comment by SquidDev** [comment, marked as answer]
It sounds like you were able to run the command correctly, just had issues with how it sounded?

CC: Tweaked's speakers can't play arbitrary audio files (such as mp3s or ogg). It has to be in a special format called DFPWM - you can convert a file to DFPWM using https://music.madefor.cc/ or https://remote.craftos-pc.cc/music/, and then should be able to download and play that!

**Reply by hay-dude007** [reply]
Thank you for giving me the URL to the website. I downloaded the file I converted, but i cant seem to get it playing in the speaker. It just tells me there is no such file

**Reply by hay-dude007** [reply]
Okay i solved the issue thanks anyways

**Reply by r4r511** [reply]
> Okay i solved the issue thanks anyways

I encounted the same problem, how did you solve it?

**Reply by fatboychummy** [reply]
How are you downloading the file (i.e: what is the exact command you are running to download it)?

**Reply by r4r511** [reply]
> How are you downloading the file (i.e: what is the exact command you are running to download it)?

Here are my code:

```lua
local dfpwm = require("cc.audio.dfpwm")
local speaker = peripheral.find("speaker")

local decoder = dfpwm.make_decoder()
for chunk in io.lines("data/example.dfpwm", 16 * 1024) do
    local buffer = decoder(chunk)

    while not speaker.playAudio(buffer) do
        os.pullEvent("speaker_audio_empty")
    end
end
```

I have already placed the file "1.dfpwm"  in the same directory as the lua file, but it still printed "no such file" on the screen when I run lua file.
How can I solve it?

**Reply by SquidDev** [reply]
You need to replace `data/example.dfpwm` with the actual name of the Lua file (it sounds like `1.dfpwm` in this case).

**Reply by r4r511** [reply]
> You need to replace `data/example.dfpwm` with the actual name of the Lua file (it sounds like `1.dfpwm` in this case).

Which directory should I put the audio files(1.dfpwm) in?
(Windows 10)

**Reply by SquidDev** [reply]
Both files should be placed in the computer's directory. This can be found by:

 1. Go to the single player menu and highlight your world.
 2. Press the "Edit" button on the bottom, then "Open World Folder"
 3. Navigate to the `computercraft/computer/<id>` folder, where `<id>` is the ID of the computer.
 4. Place the files in there.

**Reply by r4r511** [reply]
> Both files should be placed in the computer's directory. This can be found by:
> 
> 1. Go to the single player menu and highlight your world.
> 2. Press the "Edit" button on the bottom, then "Open World Folder"
> 3. Navigate to the `computercraft/computer/<id>` folder, where `<id>` is the ID of the computer.
> 4. Place the files in there.

I have solved it using your method, thank you so much for help!
